2017/11/29/systemd-primer-from-the-trenches.html.md

```diff
@@ -0,0 +1,514 @@
+pick this one up. Good April fools day. ‘’unmask’’ to return it to its user
+
+20.
+```systemctl edit serviceName```
+
+Good flags --full
+
+```vim /etc/inittab```
+
+Yes that’s right! Edit the service file without having to cd into the directory. Now that saved me a bit of time.
+
+Tip:
+* Careful with this. The plain edit creates an override file in /etc/systemd/system to complement the original unit file. 
+* If you need to edit the original unit file use the ```--full``` flag.
+* I made mistake in my unit file or I messed up a system unti file; use `systemctl revert unit`
+
+21.
+```systemctl --output=```
+
+Good Flags verbose, --full
+
+```/etc/init.d/httpd start >httpd-output 2>&1```
+
+Particular good if you have a bad service which is playing up. Outputs a short standard message or a very verbose message
+using different flags.
+
+Tip:
+* `Journalctl -u serviceName` can help you here, but I often find it easier to reply the command and include `--output=verbose`
+
+22.
+```systemctl isolate```
+
+This deserves its own small blog post. Isolate can be used to rescue systems automagically following kernel reboot 
+failures, but requires some special work.
+
+Tip:
+We will cover this next week.
+
+23.
+```systemd-delta```
+
+Check your unit files to see if someone has been changing things on you. Especially useful if you are writting your own Unit files.
+
+Tip:
+* Used in conjunction with `systemctl cat` or `edit`, `delta` can help you see what was what.
+
+## Gotchas
+
+1. Sometimes you need to use the suffix such as ‘config_file@openvpn.service’.  Keep this back of mind.
+2. systemd will always think services are services unless you use the suffix like .target or .socket, make sure you tell the system so.
+3. If you want multiple services running use a prefix, such as; ssh1@sshd.service ssh2@sshd.service with different configs. Handy for multiple openvpn servers.
+4. Mount points will always be determined as mount points.
+6. If you have made a lot of configuration changes and want to gracefully load these without restarting everything try  
+`systemctl daemon reload`
+7. This is not the same as the above “reload” action. Daemon reload is for systemd and not the unit files it controls. This is a safe 
+command to run since it keeps sockets open while it does its thing.
+8. Reboots on newer systems only really need to be done when new kernels are presented. Systemd is gracious and good at 
+processing new packages and enabling these changes during a yum update.
+9. Autocomplete on Centos is not present until you install `bash-completion`. systemctl takes on a life of it’s own, 
+when you install this utility. Tabbing out will list all systemctl options. Very handy!!
+10. Systemd does not use the /etc/inittab file even if you have them present.
+11. Converting your init’s to systemd is easier than you think. Create a systemd unit file and add in 10 basic lines to 
```

Comment:
It would be good here to give a few reasons why the reader should want to convert a SysV script that works fine under systemd, to a native systemd unit file. Perhaps mention dependencies, any other options it makes possible?

Reply:
Arh yep, was going to cover this in the UNIT post, I will elude to it here to make it clearer.

Reply:
done